environments/surfpol/configuration.py

```diff
@@ -79,8 +80,11 @@ def create_configuration_and_session(use_aws_default_profile=True, config_class=
     if aws_secrets:
         secrets_manager = session.client('secretsmanager')
         for group_name, secret_name, secret_id in aws_secrets:
-            secret_value = secrets_manager.get_secret_value(SecretId=secret_id)
-            secret_payload = json.loads(secret_value["SecretString"])
-            secrets[group_name][secret_name] = secret_payload[secret_name]
+            try:
+                secret_value = secrets_manager.get_secret_value(SecretId=secret_id)
+                secret_payload = json.loads(secret_value["SecretString"])
+                secrets[group_name][secret_name] = secret_payload[secret_name]
+            except ClientError:
```

Comment:
Moet het niet ook falen als de secret mist? Nu gaat het silent falen, weet niet of dat zo'n goed idee is 😉

Reply:
We kunnen wat logging toevoegen, maar het kan dus voorkomen dat een container geen toegang heeft tot een bepaald secret en dan faalt het. Het verbaasde mij enigszins dat de forbidden exception zo breed is als ClientError :(

Reply:
Wanneer kan dat voorkomen? Ik zou zeggen dat de container altijd toegang moet hebben tot geconfigureerde secrets. Anders moeten we overal in de applicatie er rekening mee gaan houden dat een secret mogelijk leeg kan zijn. Ik heb liever dat de applicatie dan meteen eruit klapt ipv als het secret ergens wordt gebruikt.

Reply:
We hebben dus twee verschillende database users. Eentje voor migraties/management en de ander voor al het andere. Wanneer we een standaard container runnen, dan willen we niet dat die toegang heeft tot de gehele database. Met andere woorden, we willen niet voortdurend als de owner connecten met de database. Om migration/management tasks/commands wel toegang te kunnen geven is er een secret nodig met elevated access. Deze is dan niet ingevuld voor processen in containers.